### PR TITLE
Signup: Improve transitions and fix incorrect props

### PR DESCRIPTION
**Pull request by jsnmoon**
This change mainly consists of adding animated transitions using opacity between domain step modes: the registration form, the mapping form, and the transfer form.

It also includes the following minor tweaks:
1. The About step header now ends with a period, making it consistent with headers for all other steps.
2. It removes an unused subHeaderText property from the StepWrapper in the Domain step.
3. It fixes the incorrect transition timing from the main Signup component. 

# Testing Instructions
1. Spin this up locally or in a live branch.
2. Navigate to `/start`. 
3. Fill out the form and press `Continue`. Ensure that the transitions between steps work as expected.
4. In `/start/domain`, click `Already own a domain?`. Ensure that the page transitions with an opacity animation to the transfer form.
5. In the transfer form, click `manually connect it`. Ensure that the page transitions like step 4 to the mapping form.
6. Press `Back` at the bottom of the page. Ensure that the page transitions like step 4 & 5 to the domain registration form.
7. Enter a query and select any domain name. Ensure that the page transitions as expected like step 3.

### CODE REVIEW

File: client/signup/steps/about/index.jsx [hunk 1]
```diff
@@ -576,7 +576,7 @@ class AboutStep extends Component {
 				flowName={ flowName }
 				stepName={ stepName }
 				positionInFlow={ positionInFlow }
-				headerText={ translate( 'Let’s create a site' ) }
+				headerText={ translate( 'Let’s create a site.' ) }
```
Comment (a8ci18n) [inline client/signup/steps/about/index.jsx]: :ok: This change is safe, we'll keep the existing 22 translations.

**Comment by jsnmoon** [comment]
Hey @hambai, I clearly see the fade-in transition with transfer and mapping form steps in your video. Do note that the transition between domain form steps are not the same as the transition between signup steps (ex. about -> domain selection).

**Comment by hambai** [comment]
Oh - I guess I expected the same transition as when I click on the "< Back" button.

**Comment by jsnmoon** [comment]
@sixhours @hambai 

I added a short transformation animation between domain forms that I think will address your concerns, see this [video](https://cloudup.com/cp9f1Yz5e8q).

What do you think?

**Comment by jsnmoon** [comment]
> Shouldn't we make all transitions the same? Or you made them different for the domains sections `already own/manually connect`? (@aidvu)

I omitted the exit animations due to positioning differences between our Domain form components and the StepWrapper components. As a bonus, the transitions between Domain form components is faster by 0.2s than the step transitions.

**Pull request merged**